### PR TITLE
fix: max accounts handling not respected (WPB-4818)

### PR DESCRIPTION
**Pull request by yamilmedina**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Some code were removed/adjusted and now we are not handling max account check on add "new account" self profile.

### Causes (Optional)

We can add infinte? accounts.

### Solutions

- Check for max accounts when trying to login, as a safety mechanism taking into consideration the Build flag.
- Create a proper state object for `WelcomeScreen`, so we handle them unified as in the rest of the project

### Notes (Optional)

Moving forward and not for now, since this is critical for release.
- It was discovered https://wearezeta.atlassian.net/browse/WPB-1862
- Dead code, I left some marks to refactor/remove later.
- We can unify the Dialog used on deeplinks to the General Dialog approach used here, we can migrate later.

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Attachments (Optional)

https://github.com/wireapp/wire-android-reloaded/assets/5806454/6887e31e-1396-493a-b1ec-5dac806ea72e

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/authentication/welcome/WelcomeScreen.kt [hunk 1]
```diff
@@ -161,7 +170,10 @@ private fun WelcomeContent(
                     horizontal = MaterialTheme.wireDimensions.welcomeButtonHorizontalPadding
                 )
             ) {
-                LoginButton() { navigate(NavigationCommand(LoginScreenDestination())) }
+                LoginContent(
+                    maxAccountsReached = maxAccountsReached,
+                    onLoginButtonActionClicked = { navigate(NavigationCommand(LoginScreenDestination())) }
+                )
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/authentication/welcome/WelcomeScreen.kt]: As passing `maxAccountsReached` to the login content, could it also be done for creating an account? This way if the users tries to login it gets the popup dialog, but how does the flow goes when creating a new account as we login after account creation? 🤔 just thinking of other possible ways an user can try to have more than max accounts

**Review by alexandreferris — COMMENTED** [review]
Great catch!

I just have one comment that goes by more of a concern if it is possible to break it 🤔

**Review by Garzas — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [903](https://10.10.124.134/job/AR-build-pipeline/903/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.4.0(7640911)(7640911)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.4.0(7640911)(7640911)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/authentication/welcome/WelcomeScreen.kt (hunk 1 above)
Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/ui/authentication/welcome/WelcomeScreen.kt]: good catch, solved !

**Review by gongracr — APPROVED** [review]
Good catch 🔍 It would be also great to add some automated/integrated tests for this to make sure this doesn't slip away again. GJ 💪 🍓

**Review by alexandreferris — APPROVED** [review]
👌🏻

**Comment by AndroidBob** [comment]
Build [905](https://10.10.124.134/job/AR-build-pipeline/905/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.4.0(7641240)(7641240)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.4.0(7641240)(7641240)-staging-compat.apk)

**Pull request merged**